scipy/optimize/__minpack.h

```diff
@@ -515,6 +530,7 @@ static PyObject *minpack_lmdif(PyObject *dummy, PyObject *args) {
 
  fail:
   RESTORE_FUNC();
+ fail_free:
```

Comment:
Is there a need for this actually?
There's the `callback_inited` flag already, so we can avoid adding a second label.

Reply:
I think this is necessary to avoid an infinite loop if RESTORE_FUNC fails